anndata/_core/views.py

```diff
@@ -157,6 +158,68 @@ def as_view_zappy(z, view_args):
     return z
 
 
+try:
+    from ..compat import awkward as ak
+    import weakref
+
+    # Registry to store weak references from AwkwardArrayViews to their parent AnnData container
+    _registry = weakref.WeakValueDictionary()
+    _PARAM_NAME = "_view_args"
+
+    class AwkwardArrayView(_ViewMixin, AwkArray):
```

Comment:
Subsetting the AwkwardArrayView should return a normal awkward array. Otherwise:

```python
adata = ad.AnnData(np.ones((3, 2)), obsm={"awk": ak.Array([{"a": [1, 2, 3]}, {"a": [4, 5]}, {"a": [6]}])})
v = adata[:2]

ak_view = v.obsm["awk"]

subset_of_view = ak_view[:1]
assert len(subset_of_view) == 1
subset_of_view["c"] = 1

assert not v.is_view
v.obsm["awk"].show()
```

```
[{a: [1, 2, 3], c: 1},
 {a: [4, 5], c: 1}]
```

Reply:
This seems to be a more general problem, though?

```python
adata = AnnData(np.ones((3, 2)), obsm={"arr": np.ones((3, 2))})
v = adata[:2]
arr_view = v.obsm["arr"]

subset_of_view = arr_view[:1]
assert len(subset_of_view) == 1
arr_view[:, 1] = 2

assert not v.is_view
v.obsm["arr"]
```
```
array([[1., 2.],
       [1., 2.]])
```


--- 

Related, is this intended: 

```python
adata = AnnData(np.ones((3, 2)), obsm={"arr": np.ones((3, 2))})
v = adata[:2]
arr_view = v.obsm["arr"]
arr_view[:, 1] = 2
arr_view
```
```
ArrayView([[1., 1.],
           [1., 1.]])
```
```python
v.obsm["arr"]
```
```
array([[1., 2.],
       [1., 2.]])
```

Reply:
There seem to be few quirks with AnnData views (see also https://github.com/scverse/anndata/issues/887), and in this particular issue doesn't seem to be limited to Awkward arrays. 

Can we follow up on this separately?

Reply:
From the above, the second one is intended, the first one isn't (that one is really weird). But, since the current behavior seems to be matching the behavior with numpy arrays I think it's fine to leave.